http-server/src/main/java/io/micronaut/http/server/cors/CorsFilter.java

```diff
@@ -353,6 +384,26 @@ private Optional<CorsOriginConfiguration> getConfiguration(@NonNull HttpRequest<
             .findFirst();
     }
 
+    @NonNull
+    private Optional<CorsOriginConfiguration> getAnyConfiguration(@NonNull HttpRequest<?> request) {
+        String requestOrigin = request.getOrigin().orElse(null);
+        if (requestOrigin == null) {
+            return Optional.empty();
+        }
+        for (UriRouteMatch<Object, Object> routeMatch : router.findAny(request)) {
```

Comment:
so, if we have to run the router in order to respond to the cors request, why are we changing the filter to be pre-routing at all?

Reply:
Otherwise we need to do a hack and store all the routes because it will not match